apiservers/engine/backends/container.go

```diff
@@ -284,6 +288,21 @@ func (c *Container) dockerContainerCreateParamsToPortlayer(cc types.ContainerCre
 	portLayerConfig.CreateConfig.NetworkDisabled = new(bool)
 	*portLayerConfig.CreateConfig.NetworkDisabled = cc.Config.NetworkDisabled
 
+	if !cc.Config.NetworkDisabled {
+		nc := &models.NetworkConfig{
+			NetworkName: cc.HostConfig.NetworkMode.NetworkName(),
+		}
+		if cc.NetworkingConfig != nil {
+			if es, ok := cc.NetworkingConfig.EndpointsConfig[nc.NetworkName]; ok {
+				if es.IPAMConfig != nil {
+					nc.Address = &es.IPAMConfig.IPv4Address
+				}
+			}
+		}
+
+		portLayerConfig.CreateConfig.NetworkSettings = nc
```

Comment:
Consider moving everything after the `if` to a helper.  Then we can test this path without bringing up the full daemon.